ansi/table_links.go

```diff
@@ -0,0 +1,243 @@
+package ansi
+
+import (
+	"bytes"
+	"fmt"
+	"net/url"
+	"slices"
+
+	xansi "github.com/charmbracelet/x/ansi"
+	"github.com/charmbracelet/x/exp/slice"
+	"github.com/yuin/goldmark/ast"
+	astext "github.com/yuin/goldmark/extension/ast"
+)
+
+type tableLink struct {
+	href    string
+	title   string
+	content string
+}
+
+type groupedTableLinks map[string][]tableLink
+
+type linkType int
+
+const (
+	_ linkType = iota
+	linkTypeAuto
+	linkTypeImage
+	linkTypeRegular
+)
+
+func (e *TableElement) printTableLinks(ctx RenderContext) {
+	if !e.shouldPrintTableLinks(ctx) {
+		return
+	}
+
+	w := ctx.blockStack.Current().Block
+	termWidth := int(ctx.blockStack.Width(ctx)) //nolint: gosec
+
+	renderLinkText := func(link tableLink, linkType linkType) string {
+		style := ctx.options.Styles.LinkText
+
+		var token string
+		switch linkType {
+		case linkTypeAuto:
+			token = linkWithSuffix(link, ctx.table.groupedAutoLinks)
+		case linkTypeImage:
+			token = linkWithSuffix(link, ctx.table.groupedImages)
+			style = ctx.options.Styles.ImageText
+		case linkTypeRegular:
+			token = linkWithSuffix(link, ctx.table.groupedLinks)
+		}
+
+		el := &BaseElement{Token: token, Style: style}
+		_ = el.Render(w, ctx)
+
+		return token
+	}
+
+	renderLinkHref := func(link tableLink, linkType linkType, linkText string) {
+		style := ctx.options.Styles.Link
+		if linkType == linkTypeImage {
+			style = ctx.options.Styles.Image
+		}
+
+		// TODO(@andreynering): Once #411 is merged, use the hyperlink
+		// protocol to make the link work for the full URL even if we
+		// show it truncated.
+		linkMaxWidth := max(termWidth-len(linkText)-1, 0)
+		token := xansi.Truncate(link.href, linkMaxWidth, "…")
```

Comment:
We're truncating big links with `…` to avoid polluting the result too much.

#411 is already merged to `v2-exp`, but not to `master` yet (#408).

Once merged, use here to ensure links are clickable and pointing to the full URL.